### PR TITLE
Subscription UI Changes

### PR DESCRIPTION
**Pull request by sravfeyn**
Implements https://docs.google.com/document/d/1SkdI30F1O2MbSKCEn5VSVGjse_DJUQnvmpuSiK77GN8/edit#

@nickpell @esoergel

### CODE REVIEW

File: corehq/apps/accounting/static/accounting/js/accounting.credits.js [hunk 1]
```diff
@@ -28,17 +28,31 @@ hqDefine('accounting/js/accounting.credits.js', function () {
...
+        'use strict';
+        var self = this;
+        self.name = ko.observable("Credits");
+        self.creditType = ko.observable("general_credit");
+        self.addAmount = ko.observable(0);
+        self.addAmountValid = ko.computed(function(){
+            return  parseFloat(self.addAmount()) === 0 || (parseFloat(self.addAmount()) >= 0.5);
+        });
+        self.paymentHandler = paymentHandler;
+    }
```
Comment (dimagimon) [inline corehq/apps/accounting/static/accounting/js/accounting.credits.js]: Missing semicolon. (semi)

**Comment by sravfeyn** [comment]
@nickpell I am putting this up on staging for QA. Let me know if the changes are too simple for QA and I will not proceed for a QA

**Comment by nickpell** [comment]
@sravfeyn I think an adequate way to test this is to set up several domains on staging, each of which has one combination of credit line types, and check that they are displayed correctly.  I don't think any QA beyond that is necessary.

Example: see https://staging.commcarehq.org/a/general-credit/settings/project/subscription/ and the comment I left in the doc.  This domain has credit in the general subscription credit line and no credit in any account level credit lines or any feature/product level credit lines.

**Comment by sravfeyn** [comment]
I have responded on the spec doc re. the above comment.

File: corehq/apps/domain/views.py [hunk 2]
```diff
@@ -664,6 +664,16 @@ def plan(self):
                         'renew_url': reverse(SubscriptionRenewalView.urlname, args=[self.domain]),
                     })
 
+        if subscription:
+            credit_lines = CreditLine.get_credits_by_subscription_and_features(subscription)
+            has_credits_in_non_general_credit_line = any(
```
Comment (sravfeyn) [inline corehq/apps/domain/views.py]: @nickpell This is the core of 'has credits in non general credit lines'. Can you see if it looks good?

Also per the comment on spec, should I set this to True if no credit-lines exist?

Comment (nickpell) [inline corehq/apps/domain/views.py]: I'd continue to set ```has_credits_in_non_general_credit_line``` to ```False``` if no credit lines exist

**Comment by nickpell** [comment]
Updated [comment](https://docs.google.com/document/d/1SkdI30F1O2MbSKCEn5VSVGjse_DJUQnvmpuSiK77GN8/edit?disco=AAAAA5qjvDg)

File: corehq/apps/accounting/models.py [hunk 3]
```diff
@@ -2711,6 +2711,13 @@ def get_credits_by_subscription_and_features(cls, subscription,
         ).all()
 
     @classmethod
+    def get_non_general_credits_by_subscription(cls, subscription):
```
Comment (sravfeyn) [inline corehq/apps/accounting/models.py]: @nickpell I have updated non-general credit query per your comment. Does this look right now?

File: corehq/apps/domain/views.py [hunk 4]
```diff
@@ -665,12 +665,8 @@ def plan(self):
...
 
         if subscription:
-            credit_lines = CreditLine.get_credits_by_subscription_and_features(subscription)
-            has_credits_in_non_general_credit_line = any(
-                [(line.balance > 0 and
-                 (line.product_type in SoftwareProductType.CHOICES or line.feature_type in FeatureType.CHOICES))
-                 for line in credit_lines]
-            )
+            credit_lines = CreditLine.get_non_general_credits_by_subscription(subscription)
+            has_credits_in_non_general_credit_line = len(credit_lines) > 0
```
Comment (nickpell) [inline corehq/apps/domain/views.py]: you'll still want to check that ```balance``` is greater than 0.

**Review by nickpell — CHANGES_REQUESTED** [review]
Looking close.  One comment below.  I also noticed that https://staging.commcarehq.org/a/general-credit/settings/project/subscription/ doesn't match the format in https://staging.commcarehq.org/a/no-credit/settings/project/subscription/.  Specifically, "General Credits" should not show and should be "Credits Remaining" instead.

File: corehq/apps/domain/views.py (hunk 4 above)
Comment (sravfeyn) [inline corehq/apps/domain/views.py]: Cool, I have updated this

**Comment by sravfeyn** [comment]
Re 'General Credits', there is nothing mentioned regarding this in the spec. So I haven't changed its behaviour. That text is shown `if plan.general_credit and plan.general_credit.is_visible ` per [current code](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/domain/templates/domain/current_subscription.html#L170). When should it show and not show? (I am pretty confused with 'general' vs 'non-general' so I am not able to deduct any spec logic just by looking at the code)

**Comment by sravfeyn** [comment]
@nickpell

**Comment by nickpell** [comment]
hey, I should get to this by end of wednesday at the latest

**Comment by nickpell** [comment]
make that end of thursday

**Comment by sravfeyn** [comment]
Okay

**Comment by nickpell** [comment]
Made some updates.  My test cases are as follows:

Show the new UI:
1) When you have general credit: https://staging.commcarehq.org/a/general-credit/settings/project/subscription/
2) When you have no credit: https://staging.commcarehq.org/a/no-credit/settings/project/subscription/

Show the old UI:
1) When you have plan credit: https://staging.commcarehq.org/a/plan-credit/settings/project/subscription/
2) When you have user credit: https://staging.commcarehq.org/a/feature-credit/settings/project/subscription/
4) When you have plan and user credit: https://staging.commcarehq.org/a/plan-feature-credit/settings/project/subscription/
5) When you have plan and general credit: https://staging.commcarehq.org/a/general-and-plan-credit/settings/project/subscription/
6) When you have general and user credit: https://staging.commcarehq.org/a/general-and-featurecredit/settings/project/subscription/

The Usage Summary table is blank in the new UI: https://staging.commcarehq.org/a/no-credit/settings/project/subscription/  Once you fix that and test on staging this PR will be good to merge.

@sravfeyn

**Comment by sravfeyn** [comment]
Thanks @nickpell for the updates! 

I have fixed the empty value issue. It was caused as you have referenced `limit` atttr of `CreditItem` which wasn't there. [Adding that](https://github.com/dimagi/commcare-hq/pull/14473/commits/dc5d30171e3adca94842842c968874f89e7cb7e4) fixed the issue

**Comment by nickpell** [comment]
Looks good!

**Pull request merged**